### PR TITLE
Bcrypt Hash Authentication p1

### PR DESCRIPTION
**Pull request by gdunstone**
webapp changes required for secure hash authentication see: http://php.net/manual/en/faq.passwords.php

I also changed some values in `config/db.php` to better suit the installation guide in the docs.

### CODE REVIEW

**Comment by gdunstone** [comment]
this pull request depends on https://github.com/OpenSILEX/phis-ws/pull/182

**Comment by gdunstone** [comment]
I reverted dockerization re: https://github.com/OpenSILEX/phis-ws/pull/182#issuecomment-453564383

File: controllers/SiteController.php [hunk 1]
```diff
@@ -89,13 +92,26 @@ public function actionLogin()
...
         
          if ($model->load(Yii::$app->request->post())) {
-             $model->password = password_hash($model->password, PASSWORD_BCRYPT);
-             if ($model->login()) {
-                $this->getLoggedUsersGroups();
-                return $this->goHome();
-             } else {
-                 Yii::$app->getSession()->setFlash('error', Yii::t('app/messages','Bad email / password'));
-             }
+
```
Comment (vincentmigot) [inline controllers/SiteController.php]: Can you remove unnecessary blank lines

**Comment by vincentmigot** [comment]
Hi,

Sorry not having responding you for a long time.
I tried today to integrate your pull request and actually test it.
So we can't integrate it right now because it's not working due to fact that BCrypt generate a new different hash for the same string each time you call the "password_hash" function.
You can always validate each hash with the original string but you can't validate hashes of the same string against each other and that's what you are trying to do in the webservice.

We could imagine some workarounds to make it work and we can talk about it on our Slack thread if you liked to.

But in order to be transparent with you, we plan to have a big rework on how users are managed in our software improving security and integration with other information systems :
- by getting rid of md5
- by allowing connexion against existing system with Oauth, LDAP, ...
- by sharing users between multiple OpenSilex instances

If you have other needs we will be pleased to take them in account during our rework or help you to work with us on them.

For now I will close your pull request, don't hesitate to contact us if you wan't more explaination.

Vincent

**Pull request closed without merging**